zip-0211.rst

```diff
@@ -0,0 +1,559 @@
+  * ``hashSubtreeCommitment`` is the BLAKE2b-256 hash of ``left_child || right_child``
+  * For clarity, this digest uses the BLAKE2b-256 personalization string described above.
+- serialized as ``char[32]``
+
+2. ``nEarliestTimestamp``
+
+- If the node is a leaf node
+
+  * ``nEarliestTimestamp`` is the header's timestamp
+    
+- If the node is an internal or root node
+    
+  * ``nEarliestTimestamp`` is inherited from the left child
+  
+- serialized as ``nTime`` (uint32)
+
+3. ``nLatestTimestamp``
+
+- If the node is a leaf node
+
+  * ``nLatestTimestamp`` is the header's timestamp
+  
+- If the node is an internal or root node
+  * ``nLatestTimestamp`` is inherited from the right child
+  
+- Note that due to timestamp consensus rules, ``nLatestTimestamp`` may be smaller than ``nEarliestTimestamp`` in some subtrees. This may occur within subtrees smaller than ``PoWMedianBlockSpan`` blocks.
+- serialized as ``nTime`` (uint32)
+    
+4. ``nEarliestTarget``
+
+- If the node is a leaf node
+
+  * ``nEarliestTarget`` is the header's ``nBits`` field
+  
+- If the node is an internal or root node
+  * ``nEarliestTarget`` is inherited from the left child
+  
+- serialized as ``nBits`` (uint32)
+    
+5. ``nLatestTarget``
+
+- If the node is a leaf node
+
+  * ``nLatestTarget`` is the header's ``nBits`` field
+  
+- If the node is an internal or root node
+  * ``nLatestTarget`` is inherited from the right child
+  
+- serialized as ``nBits`` (uint32)
+    
+6. ``hashEarliestSaplingRoot``
+
+- If the node is a leaf node
+
+  * ``hashEarliestSaplingRoot`` is calculated as ``hashFinalSaplingRoot``, as implemented in Sapling
+
+- If the node is an internal or root node
+
+  * ``hashEarliestSaplingRoot`` is inherited from the left child
+  
+- serialized as ``char[32]``
+    
+7. ``hashLatestSaplingRoot``
+
+- If the node is a leaf node
+
+  * ``hashLatestSaplingRoot`` is calculated as ``hashFinalSaplingRoot``, as implemented in Sapling
+
+- If the node is an internal or root node
+
+  * ``hashLatestSaplingRoot`` is inherited from the right child
+
+- serialized as ``char[32]``
+
+8. ``nEarliestHeight``
+
+- If the node is a leaf node
+
+  * ``nEarliestHeight`` is the header's height
+  
+- If the node is an internal or root node
+  * ``nEarliestHeight`` is inherited from the left child
+  
+- serialized as ``CompactSize uint``
+    
+9. ``nLatestHeight``
+
+- If the node is a leaf node
+
+  * ``nLatestHeight`` the header's height
+
+- If the node is an internal or root node
+
+  * ``nLatestHeight`` is inherited from the right child
+  * serialized as ``CompactSize uint``
+
+10. ``nSubTreeTotalWork``
+
+- If the node is a leaf node
+
+  * ``nSubTreeTotalWork`` is the protocol-defined work of the block: `floor(2 ** 256 / (toTarget(nBits) + 1))`.
+
+- If the node is an internal or root node
+
+  * ``nSubTreeTotalWork`` is the sum of the ``nSubTreeTotalWork`` fields of both children
+  
+- serialized as ``CompactSize uint``
```

Comment:
I think this needs to be larger than a `CompactSize` uint. The maximum data length of 8 bytes is not sufficient to hold the summed work of a large range of blocks as an integer. We should probably use `uint256` here.

Reply:
@therealyingtong gave me push access to the branch, so I have addressed this.